elastic/kibana
# [Synthetics] Add error handling for private locations

## Summary
Related to https://github.com/elastic/uptime/issues/475

Adds better error handling for synthetics private locations.

When adding a monitor
--
Ensures that when downstream actions related to private locations fail, the monitor saved object does not save

When editing a monitor
--
Ensures that when downstream actions related to private locations fail, the monitor is not updated

When deleting a monitor
--
Ensures that when downstream actions related to private locations fail, the monitor is not deleted

Testing
--

I tested this PR by adding deliberate errors, such as reference issues, to the `synthetics_private_location` file, including the `deleteMonitor`, `createMonitor`, and `editMonitor` code paths.

1. Add errors to the createMonitor code
2. Navigate to the Uptime, create a private location, then attempt to create a monitor with a private location
3. An error toast should appear
4. Navigate to the Uptime monitor list. Ensure the monitor was not saved.
5. Add errors to the editMonitor code
6. Navigate to the Uptime, create a monitor without a private location. It should work successfully.
7. Update that monitor by adding a new private location.  
8. An error toast should appear
9. Navigate to the Uptime monitor list. Ensure the monitor was not updated and the new location does not appear to be saved
10. Remove the error from `createMonitor` code
11. Create a monitor with a private location. It should go through successfly. 
12. Add an error to the `deleteMonitor` code. 
13. Attempt to delete the monitor with the private location. 
14. An error toast should appear. 
15. Ensure the monitor is not deleted



### Checklist

Delete any items that are not applicable to this PR.
